### PR TITLE
KEYCLOAK-2236 add service account support to keycloak-admin-client

### PR DESCRIPTION
**Pull request by grossws**
This patch adds support for `grant_type=client_credentials` to `keycloak-admin-client` so service account can be used for invoking admin methods on Keycloak.

Fixes #KEYCLOAK-2236

### CODE REVIEW

File: adapters/saml/undertow/pom.xml [hunk 1]
```diff
@@ -32,12 +32,6 @@
 
     <dependencies>
         <dependency>
-            <groupId>org.jboss.logging</groupId>
```
Comment (stianst) [inline adapters/saml/undertow/pom.xml]: This change is not related to KEYCLOAK-2236

Comment (grossws) [inline adapters/saml/undertow/pom.xml]: Yes, it was prerequisite to maven project to be valid. Should I update PR and drop this commit from it?

Comment (stianst) [inline adapters/saml/undertow/pom.xml]: Yes please. You can send a separate PR for that change if you want (no need to create JIRA).

**Comment by stianst** [comment]
Other changes to the admin client has caused merge conflicts here. Would you be able to update the PR?

**Comment by grossws** [comment]
Ok, I'll rebase it on current keycloak master today.

**Comment by grossws** [comment]
Rebased on current master and exluded commit modifying unrelated pom.

**Comment by grossws** [comment]
I had accidentally removed `ResteasyClientBuilder` [configuration](https://github.com/keycloak/keycloak/pull/2449/files#diff-6765e7e44a80665caff397cd0531b891L49). I can send another PR if you want.

**Comment by stianst** [comment]
Yes please
